### PR TITLE
redo(ticdc): support for applying ddl event in applier

### LINKED ISSUE #8361: Make redo applier support consuming DDL events

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #8361

### What is changed and how it works?
1. Support for applying ddl event in applier
2. Read and decode log events in background goroutines in logReader. 
3. [Add TableName field to RedoDDLEvent](https://github.com/pingcap/tiflow/pull/8362/files#diff-6d2f9b5931ca742dd70204701ec3baa8cce6350a61e29d55a0e85176b7618fcaR112).

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 
### Release note 

```release-note
`support for applying ddl event in redo log`.
```

### CODE REVIEW

File: cdc/redo/reader/reader.go [hunk 1]
```diff
@@ -145,203 +141,140 @@ func (l *LogReader) ResetReader(ctx context.Context, startTs, endTs uint64) erro
...
-	if err != nil {
-		return err
-	}
-
+func (l *LogReader) runDDLReader(egCtx context.Context) error {
+	defer close(l.ddlCh)
 	ddlCfg := &readerConfig{
-		startTs:            startTs,
-		endTs:              endTs,
+		startTs:            l.meta.CheckpointTs - 1,
```
Comment (nongfushanquan) [inline cdc/redo/reader/reader.go:171]: why is -1 needed?

Comment (CharlesCheung96) [reply cdc/redo/reader/reader.go:171]: If there is a ddl event in the check point, there is no way to determine whether the event was executed. Therefore, it is a safe practice to always assume that the event did not execute successfully and try to execute this DDL event in applier, which is similar to initializing DDLPuller with `checkpointTs - 1` when owner restarts.

File: pkg/applier/main_test [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Copyright 2023 PingCAP, Inc.
```
Comment (amyangfei) [inline pkg/applier/main_test:1]: Filename should be `main_test.go`?

File: pkg/applier/redo.go [hunk 3]
```diff
@@ -103,101 +117,207 @@ func (ra *RedoApplier) catchError(ctx context.Context) error {
 	}
 }
 
+func (ra *RedoApplier) initSink(ctx context.Context) (err error) {
+	// MySQL sink will use the following replication config
+	// - EnableOldValue: default true
+	// - ForceReplicate: default false
+	// - filter: default []string{"*.*"}
```
Comment (amyangfei) [inline pkg/applier/redo.go:124]: Seems these comments are not necessary and can remove them.

File: cdc/redo/reader/reader.go [hunk 4]
```diff
@@ -22,30 +22,30 @@ import (
...
-	// The returned redo logs sorted by commit-ts
-	ReadNextLog(ctx context.Context, maxNumberOfEvents uint64) ([]*model.RowChangedEvent, error)
-
-	// ReadNextDDL reads `maxNumberOfDDLs` ddl events from redo logs from current cursor
-	ReadNextDDL(ctx context.Context, maxNumberOfEvents uint64) ([]*model.DDLEvent, error)
-
+	// Run read and decode redo logs in background.
+	Run(ctx context.Context) error
+	// ReadNextRow read one row event from redo logs.
+	ReadNextRow(ctx context.Context) (*model.RowChangedEvent, error)
```
Comment (amyangfei) [inline cdc/redo/reader/reader.go:46]: Before this pr, what is the batch size that `ReadNextLog` return in most cases? Is there any performance issue that changing it from batch to one row?

Comment (CharlesCheung96) [reply cdc/redo/reader/reader.go:46]: Currently we use [buffered channel](https://github.com/pingcap/tiflow/pull/8362/files#diff-7194f7523941d867fef7ed626c77db0337996b8bc35063b9fa595099215f51beR108) to cache logs with [the same capacity](https://github.com/pingcap/tiflow/pull/8362/files#diff-7194f7523941d867fef7ed626c77db0337996b8bc35063b9fa595099215f51beR38) as the previous batchsize.

**Review by amyangfei — APPROVED** [review]

**Pull request merged**